generate.go

```diff
@@ -85,6 +85,7 @@ func (zp *ZoneParser) generate(l lex) (RR, bool) {
 	}
 	zp.sub = NewZoneParser(r, zp.origin, zp.file)
 	zp.sub.includeDepth, zp.sub.includeAllowed = zp.includeDepth, zp.includeAllowed
+	zp.sub.SetGenerateAllowed(false)
```

Comment:
I don’t think we should add any exported API for this, just a direct `zp.sub.generateAllowed = false` should work fine.